### DISCUSSION TITLE
Building for qemu v6.1.0

### DISCUSSION THREAD

**Discussion by jonathan-albrecht-ibm** [tonistiigi/binfmt#57] [General]
I tried building binfmt against the newly released qemu v6.1.0 but hit a compile error. @crazy-max it's the same error as seen in the recent release GitHub action https://github.com/tonistiigi/binfmt/actions/runs/1164593346

I just wanted to note that I tried building without applying the alpine aport qemu patches (https://github.com/tonistiigi/binfmt/blob/99c76af6d6aec693ab9195190055c0262d95d667/Dockerfile#L20) and everything was successful. It looks like there were some fixes in qemu since v6.0.0 to help compiling against musl. I think its the https://github.com/alpinelinux/aports/blob/master/community/qemu/fix-sigevent-and-sigval_t.patch that now causes the compilation error but I haven't looked to closely.

I don't know if any of the other patches are needed for other reasons though.

Hope that helps.

**Comment by crazy-max** [comment]
@jonathan-albrecht-ibm Yes indeed we have discussed about this with @tonistiigi and it should work with the macro so we can remove this patch now.

**Reply by jonathan-albrecht-ibm** [reply]
Good to hear, thx @crazy-max @tonistiigi

**Comment by tonistiigi** [comment]
Test build is up in `tonistiigi/binfmt:qemu-v6.1.0` https://hub.docker.com/r/tonistiigi/binfmt/tags?page=1&ordering=last_updated&name=v6.1.0 https://github.com/tonistiigi/binfmt/actions/runs/1171279112 built from #58

**Reply by jonathan-albrecht-ibm** [reply]
Thx @tonistiigi, I tried the `tonistiigi/binfmt:qemu-v6.1.0` image on but I'm seeing some problems with networking when running an s390x/ubuntu image.
```
$ docker run --privileged --rm tonistiigi/binfmt:qemu-v6.1.0 --install all
... install output looks good ...
$ docker run --rm -it --platform=s390x s390x/ubuntu "/bin/bash"
root@4a679ca9d169:/# apt update
Err:1 http://ports.ubuntu.com/ubuntu-ports focal InRelease
  Temporary failure resolving 'ports.ubuntu.com'
... more of the same, update failed ...
```
but if I build my own test image with glibc from commit e7205b99a31dda065c25c4f1a41c70352c436fe2:
```
$ docker run --privileged --rm tonistiigi/binfmt:qemu-v6.1.0 --uninstall qemu-*
$ git checkout e7205b99a31dda065c25c4f1a41c70352c436fe2
$ QEMU_VERSION=v6.1.0  docker buildx bake --load mainline --no-cache
$ docker run --privileged --rm tonistiigi/binfmt:test --install all
... install output looks good ...
$ $ docker run --rm -it --platform=s390x s390x/ubuntu "/bin/bash"
root@3f858a99b24c:/# apt update
Get:1 http://ports.ubuntu.com/ubuntu-ports focal InRelease [265 kB]
... more of the same, update succeeded ...
```
Not sure what's happening here.

This is my last day before vacation so I might not be able to respond quickly until Sept 7

**Reply by jonathan-albrecht-ibm** [reply]
I also get the same behavior when running arm64 `docker run --rm -it --platform=arm64 arm64v8/ubuntu "/bin/bash"`

**Reply by jonathan-albrecht-ibm** [reply]
Did some googling and found this: https://github.com/ziglang/zig/pull/8750#issuecomment-842096111 with a link to a patch that was submitted to qemu but not yet merged. 

I applied it to my local qemu repo and built binfmt against it on #58 and it seems to fix the problems above and I haven't hit any other issues so far.

**Reply by tonistiigi** [reply]
Yes, found the same thing https://github.com/void-linux/void-packages/commit/62ca689e155f5cee7506ed3661e92dfcf97ebca0#diff-0cc48ed3411d25c1bb524b321ca82ff9abf251bbbd1f1050360b49e4862f524c . Will try to apply it to #58

**Reply by tonistiigi** [reply]
new build https://github.com/tonistiigi/binfmt/actions/runs/1175497869

**Reply by jonathan-albrecht-ibm** [reply]
Back now and tried the new build with an s390x docker container. It looks good. Haven't hit any problems. Thanks @tonistiigi @crazy-max
